prow/plugins/approve_configtree_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
+
+import (
+	"testing"
+
+	"sigs.k8s.io/yaml"
+)
+
+func TestApproveConfigTree(t *testing.T) {
+	var cases = []struct {
+		name        string
+		config      []byte
+		expectError bool
+	}{
+		{
+			name: "approve no orgs",
+			config: []byte(`
+config:
+  issue_required: false
+  require_self_approval: true
+  lgtm_acts_as_approve: false
+  ignore_review_state: true
+`),
+			expectError: true,
+		},
+		{
+			name: "approve no default",
+			config: []byte(`
+orgs:
+  bazelbuild:
+    config:
+      ignore_review_state: false
+  kubernetes:
+    config:
+      lgtm_acts_as_approve: true
+    repos:
+      kops:
+        config:
+          lgtm_acts_as_approve: false
+      kubernetes:
+        config:
+          require_self_approval: true
+`),
+		},
+		{
+			name: "approve full",
+			config: []byte(`
+config:
+  issue_required: false
+  require_self_approval: false
+  lgtm_acts_as_approve: false
+  ignore_review_state: true
+orgs:
+  bazelbuild:
+    config:
+      ignore_review_state: false
+  kubernetes:
+    config:
+      lgtm_acts_as_approve: true
+    repos:
+      kops:
+        config:
+          lgtm_acts_as_approve: false
+      kubernetes:
+        config:
+          require_self_approval: true
+        branches:
+          master:
+            require_self_approval: false
+`),
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			var tree ApproveConfigTree
+			if err := yaml.Unmarshal(c.config, &tree); err != nil {
+				t.Errorf("error unmarshaling config: %v", err)
+			}
+			rsa := tree.BranchOptions("kubernetes", "kubernetes", "dev").RequireSelfApproval
```

Comment:
If we decide to keep these tests please test the `OrgOptions` and `RepoOptions` functions and check that all the fields actually have the expected values.